IntelFsp2Pkg/Tools/ConfigEditor/GenYamlCfg.py

```diff
@@ -730,7 +732,12 @@ def format_struct_field_name(input, count=0):
 
         if count > 1:
             name = '%s[%d]' % (name, count)
-
+        #
+        # FSP[T/M/S]_ARCH2_UPD struct field name is Fsp[t/m/s]Arch2Upd,
+        # it should change to Fsp[t/m/s]ArchUpd for code compatibility.
+        #
+        if re.match('Fsp[tms]Arch2Upd', name):
+            name = re.sub(r'\d+', '', name)

```

Comment:
It's a simple removing-"2" task. But the code is implemented in a too complicated way.

Reply:
code is updated